components/swim/src/server/mod.rs

```diff
@@ -0,0 +1,185 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub mod inbound;
+pub mod outbound;
+
+use std::collections::HashSet;
+use std::clone::Clone;
+use std::fmt;
+use std::net::{ToSocketAddrs, UdpSocket, SocketAddr};
+use std::sync::{Arc, RwLock};
+use std::sync::atomic::{AtomicBool, AtomicIsize, Ordering};
+use std::sync::mpsc::channel;
+use std::time::Duration;
+use std::thread;
+
+use error::Result;
+use member::{Member, MemberList};
+use trace::Trace;
+
+#[derive(Debug)]
+pub struct Server {
+    pub name: Arc<RwLock<String>>,
+    pub socket: UdpSocket,
+    pub member: Arc<RwLock<Member>>,
+    pub member_list: Arc<RwLock<MemberList>>,
+    pub pause: Arc<AtomicBool>,
+    pub trace: Arc<RwLock<Trace>>,
+    pub rounds: Arc<AtomicIsize>,
+    pub blacklist: Arc<RwLock<HashSet<SocketAddr>>>,
+}
+
+impl Clone for Server {
+    fn clone(&self) -> Server {
+        let socket = match self.socket.try_clone() {
+            Ok(socket) => socket,
+            Err(e) => {
+                println!("Failed to clone socket; trying again: {:?}", e);
+                return self.clone();
+            },
+        };
+        Server {
+            name: self.name.clone(),
+            socket: socket,
+            member: self.member.clone(),
+            member_list: self.member_list.clone(),
+            pause: self.pause.clone(),
+            trace: self.trace.clone(),
+            rounds: self.rounds.clone(),
+            blacklist: self.blacklist.clone(),
+        }
+    }
+}
+
+impl Server {
+    pub fn new<A: ToSocketAddrs>(addr: A, member: Member, trace: Trace) -> Result<Server> {
+        let socket = try!(UdpSocket::bind(addr));
+        try!(socket.set_read_timeout(Some(Duration::from_millis(1000))));
+        try!(socket.set_write_timeout(Some(Duration::from_millis(1000))));
+        try!(socket.set_nonblocking(false));
+        Ok(Server {
+            name: Arc::new(RwLock::new(String::from(member.get_id()))),
+            socket: socket,
+            member: Arc::new(RwLock::new(member)),
+            member_list: Arc::new(RwLock::new(MemberList::new())),
+            pause: Arc::new(AtomicBool::new(false)),
+            trace: Arc::new(RwLock::new(trace)),
+            rounds: Arc::new(AtomicIsize::new(0)),
+            blacklist: Arc::new(RwLock::new(HashSet::new())),
+        })
+    }
+
+    pub fn rounds(&self) -> isize {
+        self.rounds.load(Ordering::SeqCst)
+    }
+
+    pub fn update_round(&self) {
+        let current_round = self.rounds.load(Ordering::SeqCst);
+        match current_round.checked_add(1) {
+            Some(_number) => {
+                self.rounds.fetch_add(1, Ordering::SeqCst);
+                return
```

Comment:
Explicit call to `return` here is unncessary, you'll drop out of the match and return `()`